tests/main.test.ts

```diff
@@ -1709,3 +1709,81 @@ describe(`handle integrity checks`, () => {
     });
   });
 });
+
+it(`should allow range versions in devEngines.packageManager.version when user specifies exact npm version`, async () => {
```

Comment:
nit: we could tune down the WETness here
```suggestion
describe(`allow range versions in devEngines.packageManager.version when user specifies exact version`, () => {
  it(`should work with npm`, async () => {
```
Or even use a loop
```suggestion
describe(`allow range versions in devEngines.packageManager.version when user specifies exact version`, () => {
  for (const { name, versionRange, userProvidedVersion } of [
    { name: `npm`, versionRange: `^10.7.0`, userProvidedVersion: `6.14.2` },
    { name: `yarn`, versionRange: `^2.2.0`, userProvidedVersion: `2.2.2` },
    { name: `pnpm`, versionRange: `^5.8.0`, userProvidedVersion: `5.8.0` },
  ]) {
    it(`should work with ${name}`, async () => {
```